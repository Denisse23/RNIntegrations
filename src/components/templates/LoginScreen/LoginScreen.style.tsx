import styled from '@emotion/native';
import { ContainerWithCenterJustify } from '@styles/SharedStyles';
import { scaleBySize } from '@utils/Scale';

export const StyledContainer = styled(ContainerWithCenterJustify)(
  ({ theme }) => ({
    marginTop: scaleBySize(100),
  }),
);
