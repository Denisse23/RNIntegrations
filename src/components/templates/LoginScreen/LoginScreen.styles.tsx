import styled from '@emotion/native';
import { ContainerWithCenterJustify } from '@styles/SharedStyles';
import { scaleBySize } from '@utils/Scale';

export const StyledContainer = styled(ContainerWithCenterJustify)(() => ({
  marginTop: scaleBySize(100),
}));
