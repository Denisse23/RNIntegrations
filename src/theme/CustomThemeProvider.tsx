import React from 'react';
import { useColorScheme } from 'react-native';
import { ThemeProvider } from '@emotion/react';
import { ColorScheme, DarkTheme, LightTheme } from './Themes';

const CustomThemeProvider = ({ children }: { children: React.ReactNode }) => {
  const scheme = useColorScheme();

  return (
    <ThemeProvider theme={scheme === ColorScheme.DARK ? DarkTheme : LightTheme}>
      {children}
    </ThemeProvider>
  );
};

export default CustomThemeProvider;
