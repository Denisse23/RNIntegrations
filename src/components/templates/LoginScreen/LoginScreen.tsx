import React, { useState } from 'react';
import { Button, CustomInput, CustomText } from '@components/atoms';
import { FormLayout } from '@components/organisms';
import { NavigationType } from '@navigation/types';
import { useNavigation } from '@react-navigation/native';
import { Divider, LargeDivider } from '@styles/SharedStyles';
import { BUILD_ENV } from 'react-native-config';
import { INPUT_COMMON_PROPS, TEST_IDS } from '@constants';
import { useAuth } from '@hooks';
import strings from '@localization';
import { StyledContainer } from './LoginScreen.styles';

const LoginScreen = () => {
  const navigation = useNavigation<NavigationType>();

  const { onLoginPress } = useAuth();

  const [emailValue, setEmailValue] = useState('');
  const [passwordValue, setPasswordValue] = useState('');
  const [emailError, setEmailError] = useState<string | null>(null);
  const [passwordError, setPasswordError] = useState<string | null>(null);

  const loginDisabled = () => {
    return !emailValue || !passwordValue || !!emailError || !!passwordError;
  };

  return (
    <FormLayout testID={TEST_IDS.templates.signInScreen}>
      <StyledContainer>
        <CustomText size="xxl" type="bold">
          {`${strings.authentication.login} ${BUILD_ENV}`}
        </CustomText>
        <LargeDivider />
        <CustomInput
          {...INPUT_COMMON_PROPS.email}
          label={strings.authentication.email}
          value={emailValue}
          onChangeText={setEmailValue}
          error={emailError}
          onChangeError={setEmailError}
          testID={TEST_IDS.inputs.emailInput}
        />
        <Divider />
        <CustomInput
          {...INPUT_COMMON_PROPS.password}
          label={strings.authentication.password}
          value={passwordValue}
          onChangeText={setPasswordValue}
          error={passwordError}
          onChangeError={setPasswordError}
          testID={TEST_IDS.inputs.passwordInput}
        />
        <LargeDivider />
        <LargeDivider />
        <Button
          text={strings.authentication.login}
          type="primary"
          disabled={loginDisabled()}
          testID={TEST_IDS.buttons.signInButton}
          onPress={onLoginPress}
        />
        <Button
          text={strings.authentication.notAccount}
          type="primaryText"
          onPress={() => navigation.navigate('SignUpScreen')}
          testID={TEST_IDS.buttons.notAccountButton}
        />
      </StyledContainer>
    </FormLayout>
  );
};

export default LoginScreen;
