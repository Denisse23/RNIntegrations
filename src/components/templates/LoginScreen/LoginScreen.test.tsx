import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import CustomThemeProvider from '@theme/CustomThemeProvider';
import renderer from 'react-test-renderer';
import { TEST_IDS } from '@constants';
import LoginScreen from './LoginScreen';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

describe('LoginScreen', () => {
  it('should not enable button if the inputs are empty', async () => {
    const { getByTestId } = render(
      <CustomThemeProvider>
        <LoginScreen />
      </CustomThemeProvider>,
    );

    const emailInput = getByTestId(TEST_IDS.inputs.emailInput);
    await waitFor(() => {
      fireEvent.changeText(emailInput, 'test@gmail.com');
    });

    await waitFor(() => {
      fireEvent.changeText(emailInput, 'test@gmail.com');
    });

    const button = getByTestId(TEST_IDS.buttons.signInButton);

    expect(button.props).toHaveProperty('accessibilityState.disabled', true);
  });

  it('should enable button if the inputs are not empty', async () => {
    const { getByTestId } = render(
      <CustomThemeProvider>
        <LoginScreen />
      </CustomThemeProvider>,
    );

    const emailInput = getByTestId(TEST_IDS.inputs.emailInput);
    const passwordInput = getByTestId(TEST_IDS.inputs.passwordInput);
    await waitFor(() => {
      fireEvent.changeText(emailInput, 'test@gmail.com');
      fireEvent.changeText(passwordInput, 'Test12345!');
    });

    await waitFor(() => {
      fireEvent.changeText(emailInput, 'test@gmail.com');
    });

    const button = getByTestId(TEST_IDS.buttons.signInButton);

    expect(button.props).toHaveProperty('accessibilityState.disabled', false);
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(
      <CustomThemeProvider>
        <LoginScreen />
      </CustomThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
