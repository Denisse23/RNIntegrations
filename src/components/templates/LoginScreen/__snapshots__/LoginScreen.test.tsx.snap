// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should match the snapshot 1`] = `
<RCTSafeAreaView
  accessibilityLabel="sign-in-screen"
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
  testID="sign-in-screen"
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Array [
              Object {
                "padding": 37.5,
              },
              undefined,
            ],
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "justifyContent": "center",
              },
              Array [
                Object {
                  "marginTop": 234.375,
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            size="xxl"
            style={
              Array [
                Object {
                  "color": "#353434",
                  "fontSize": 46.875,
                  "fontWeight": "700",
                },
                undefined,
              ]
            }
            type="bold"
          >
            Login undefined
          </Text>
          <View
            style={
              Array [
                Object {
                  "height": 75,
                },
                undefined,
              ]
            }
          />
          <View
            error={null}
            style={
              Array [
                Object {
                  "backgroundColor": "#D5D5D533",
                  "borderBottomWidth": 2.34375,
                  "borderColor": "#D5D5D5",
                  "borderLeftWidth": 7.03125,
                  "borderRadius": 75,
                  "borderRightWidth": 2.34375,
                  "borderTopWidth": 7.03125,
                  "height": 117.1875,
                  "paddingHorizontal": 18.75,
                },
                undefined,
              ]
            }
          >
            <TextInput
              accessibilityLabel="email-input"
              autoCapitalize="none"
              keyboardType="email-address"
              maxLength={100}
              onChangeText={[Function]}
              placeholder="Email"
              placeholderTextColor="#A9A8A3"
              style={
                Array [
                  Object {
                    "color": "#353434",
                    "fontSize": 32.8125,
                    "height": "100%",
                    "paddingHorizontal": 43.9453125,
                    "paddingTop": 0,
                    "position": "absolute",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
              testID="email-input"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "height": 37.5,
                },
                undefined,
              ]
            }
          />
          <View
            error={null}
            style={
              Array [
                Object {
                  "backgroundColor": "#D5D5D533",
                  "borderBottomWidth": 2.34375,
                  "borderColor": "#D5D5D5",
                  "borderLeftWidth": 7.03125,
                  "borderRadius": 75,
                  "borderRightWidth": 2.34375,
                  "borderTopWidth": 7.03125,
                  "height": 117.1875,
                  "paddingHorizontal": 18.75,
                },
                undefined,
              ]
            }
          >
            <TextInput
              accessibilityLabel="password-input"
              autoCapitalize="none"
              keyboardType="default"
              multiline={false}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="#A9A8A3"
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#353434",
                    "fontSize": 32.8125,
                    "height": "100%",
                    "paddingHorizontal": 43.9453125,
                    "paddingTop": 0,
                    "position": "absolute",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
              testID="password-input"
              textContentType="password"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "height": 75,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "height": 75,
                },
                undefined,
              ]
            }
          />
          <View
            accessibilityLabel="sign-in-button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F5BE59",
                "borderRadius": 75,
                "height": 107.8125,
                "justifyContent": "center",
                "opacity": 0.6,
              }
            }
            testID="sign-in-button"
          >
            <Text
              size="md"
              style={
                Array [
                  Object {
                    "color": "#353434",
                    "fontSize": 32.8125,
                    "fontWeight": "700",
                  },
                  Object {
                    "color": "#353434",
                  },
                ]
              }
              type="bold"
            >
              Login
            </Text>
          </View>
          <View
            accessibilityLabel="not-account-button"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FAFAFA",
                "borderRadius": 75,
                "height": 107.8125,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID="not-account-button"
          >
            <Text
              size="md"
              style={
                Array [
                  Object {
                    "color": "#353434",
                    "fontSize": 32.8125,
                    "fontWeight": "700",
                  },
                  Object {
                    "color": "#A9A8A3",
                  },
                ]
              }
              type="bold"
            >
              Don't have an account yet? Sign up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
